app/controllers/api/mixins/genealogy.rb

```diff
@@ -0,0 +1,58 @@
+module Api
+  module Mixins
+    module Genealogy
+      RELATIONSHIP_COLLECTIONS = %w[vms templates].freeze
+      VALID_EDIT_ATTRS = %w[description name child_resources parent_resource].freeze
```

Comment:
This is interesting...I'm wondering if we leave the valid attrs in the individual models and only "add" child_resources  and parent_resource...I'm ok with this being in a follow up though, since it might be fairly complicated?

Reply:
I think the way to do this "right" is to have a VmsAndTemplates mixin.  In there we define `VALID_EDIT_ATTRS = %w[description name].freeze`, and remove it from here, and then also move `valid_custom_attrs` into VmsAndTemplate Mixin.  Finally, in the VmsAndTemplates mixin, you add in an include of Api::Mixins::Genealogy (alternatively just inline it right into the VmsAndTemplates mixin).

Then we add here: `VALID_GENEALOGY_ATTRS = %w[child_resources parent_resource].freeze`

Then down below we do 

```
        def validate_edit_data(data)
-         invalid_keys = data.keys - VALID_EDIT_ATTRS - valid_custom_attrs
+        invalid_keys = data.keys - VALID_EDIT_ATTRS - VALID_GENEALOGY_ATTRS - valid_custom_attrs
```

Something like that.

Reply:
I can push a commit with my idea if that's what we want to do

Reply:
I think that sounds good. If you want you can push a commit I can also try to do this on my end if you are busy.

Reply:
@Fryguy I pushed a commit with all the other suggestions, if you want I can also try implementing this one as well.

Reply:
@Fryguy I tried to implement this and it just broke my code. I'm assuming I did something wrong and not exactly sure which methods need to go in which mixin.

Reply:
I do like the idea of having the decision of what keys are valid as part of the individual controller. nice call